packages/blockchain/src/index.ts

```diff
@@ -109,17 +110,17 @@ export interface BlockchainOptions {
   genesisBlock?: Block
 
   /**
-   * Optional consensus class. If not provided, this will default to the consensus
-   * class determined by the consensus algorithm set in the {@link Common} instance.
+   * Optional consensus algorithm. If not provided, this will default to the
+   * consensus algorithm set in the {@link Common} instance.
    */
-  consensus?: ConsensusBase
+  consensus?: ConsensusAlgorithm
 }
 
 /**
  * This class stores and interacts with blocks.
  */
 export default class Blockchain implements BlockchainInterface {
-  consensus: ConsensusBase | CliqueConsensus | EthashConsensus
+  consensus: Consensus | CasperConsensus | CliqueConsensus | EthashConsensus
```

Comment:
This here should only be typed with the main `Consensus` interface, even if there are some additional methods. This is one of the goals of this refactoring to keep this as unified as possible. If there *remains* some need to access e.g. `CasperConsensus` methods, there should be some dedicated cast on the call location itself.

Reply:
Updated in new PR